droidlet/lowlevel/hello_robot/remote/obstacle_utils.py

```diff
@@ -79,12 +82,44 @@ def is_obstacle(
         raise RuntimeError("Not Implemented")
     elif num_cropped_points >= pix_threshold:
         rest = get_ground_plane(crop, return_ground=False)
-        obstacle = np.asarray(rest.points).shape[0] > 100
+        if np.asarray(rest.points).shape[0] > 100:
+            obstacle = True
+
+    if lidar is not None:
+        if is_lidar_obstacle(lidar):
+            obstacle = True
+
     if return_viz:
         return obstacle, pcd, crop, bbox, rest
     else:
         return obstacle
 
+def is_lidar_obstacle(lidar_scan, bbox=(0.0, 0.5, -0.25, 0.25), min_quality=0):
+    # bbox specifies coordinates of rectangle (xmin, xmax, ymin, ymax) centered
+    # at lidar. Any points within this box are considered an obstacle.
+    # Note that positive x is the front of the robot.
+    timestamp, scan = lidar_scan
+
+    age = time.time() - timestamp  # in seconds
+    if age > 1.0:
+        warnings.warn(f"[is_lidar_obstacle] lidar scan is {age:.2f} seconds old")
```

Comment:
did you see this get triggered in practice?

Reply:
No, I never saw it